modules/hcp-bm-add-nodes-to-inventory.adoc

```diff
@@ -0,0 +1,310 @@
+// Module included in the following assemblies:
+//
+// * hosted_control_planes/hcp-deploy/hcp-deploy-bm.adoc
+
+
+:_mod-docs-content-type: PROCEDURE
+[id="hcp-bm-add-nodes-to-inventory_{context}"]
+= Creating an infrastructure environment and adding nodes
+
+On {hcp}, the control-plane components run as pods on the management cluster while the data plane runs on dedicated nodes. You can use the Assisted Service to boot your hardware with a discovery ISO that adds your hardware to a hardware inventory. Later, when you create a hosted cluster, the hardware from the inventory is used to provision the data-plane nodes. The object that is used to get the discovery ISO is an `InfraEnv` resource. You need to create a `BareMetalHost` object that configures the cluster to boot the bare-metal node from the discovery ISO.
+
+.Procedure
+
+. Create a namespace to store your hardware inventory by entering the following command:
++
+[source,terminal]
+----
+$ oc --kubeconfig ~/<directory_example>/mgmt-kubeconfig create \
+  namespace <namespace_example>
+----
++
+where:
++
+<directory_example>:: Is the name of the directory where the kubeconfig file for the management cluster is saved.
+<namespace_example>:: Is the name of the namespace that you are creating; for example, `hardware-inventory`.
++
+.Example output
+[source,terminal]
+----
+namespace/hardware-inventory created
+----
+
+. Copy the pull secret of the management cluster by entering the following command:
++
+[source,terminal]
+----
+$ oc --kubeconfig ~/<directory_example>/mgmt-kubeconfig \
+  -n openshift-config get secret pull-secret -o yaml \
+  | grep -vE "uid|resourceVersion|creationTimestamp|namespace" \
+  | sed "s/openshift-config/<namespace_example>/g" \
+  | oc --kubeconfig ~/<directory_example>/mgmt-kubeconfig \
+  -n <namespace> apply -f -
+----
++
+where:
++
+<directory_example>:: Is the name of the directory where the kubeconfig file for the management cluster is saved.
+<namespace_example>:: Is the name of the namespace that you are creating; for example, `hardware-inventory`.
++
+.Example output
+[source,terminal]
+----
+secret/pull-secret created
+----
+
+. Create the InfraEnv resource by adding the following content to a YAML file:
++
+[source,yaml]
+----
+apiVersion: agent-install.openshift.io/v1beta1
+kind: InfraEnv
+metadata:
+  name: hosted
+  namespace: <namespace_example>
+spec:
+  additionalNTPSources:
+  - <ip_address>
+  pullSecretRef:
+    name: pull-secret
+  sshAuthorizedKey: <ssh_public_key>
```

Comment:
I would add a couple of "..." under the last line, just to let the user know that there are some other options not mentioned in the sample.